synapse/rest/client/v1/room.py

```diff
@@ -1389,6 +1389,34 @@ def register_txn_path(servlet, regex_string, http_server, with_get=False):
         )
 
 
+class TimestampLookupRestServlet(RestServlet):
+    PATTERNS = client_patterns("/rooms/(?P<room_id>[^/]*)/timestamp_to_event$")
+
+    def __init__(self, hs: "HomeServer"):
+        super().__init__()
+        self._auth = hs.get_auth()
+        self._store = hs.get_datastore()
+
+    async def on_GET(
+        self, request: SynapseRequest, room_id: str
+    ) -> Tuple[int, JsonDict]:
+        requester = await self._auth.get_user_by_req(request)
+        await self._auth.check_joined_room(room_id, requester.to_string())
+
+        timestamp = parse_integer(request, "ts")
+        thread_id = parse_integer(request, "thread_id", 0)
```

Comment:
What is a `thread_id`?

Reply:
It looks like it was part of a threading demo and we can remove it here, https://gitlab.matrix.org/new-vector/synapse/-/commit/e6896040c71f56aa59d8f8a2d1e788d0ad79b4ac